src/scanner.c

```diff
@@ -0,0 +1,44 @@
+#include <tree_sitter/parser.h>
+#include <wctype.h>
+#include <stdio.h>
+
+enum TokenType {
+  TYPE_TOKEN
+};
+
+void *tree_sitter_jsdoc_external_scanner_create() { return NULL; }
+void tree_sitter_jsdoc_external_scanner_destroy(void *p) {}
+void tree_sitter_jsdoc_external_scanner_reset(void *p) {}
+unsigned tree_sitter_jsdoc_external_scanner_serialize(void *p, char *buffer) { return 0; }
+void tree_sitter_jsdoc_external_scanner_deserialize(void *p, const char *b, unsigned n) {}
+
+// Scan to the next balanced `}` character.
+bool scan_for_type(TSLexer *lexer) {
+  int stack = 0;
+  while (true) {
+    switch (lexer->lookahead) {
+      case '{':
+        stack++;
+        break;
+      case '}':
+        stack--;
+        if (stack == -1) { return true; }
+        break;
+      case '\n':
+        // Something's gone wrong.
+        return false;
+      default:;
```

Comment:
Can we handle the eof or null case? It's possible we might get stuck in an infinite loop here, might also be a good idea to pull in the fuzz action

Reply:
I can certainly handle eof, but the fuzz action is outside of my expertise.

Reply:
Here's an example - just plop this in .github/workflows/fuzz.yml and change the name to jsdoc
https://github.com/tree-sitter/tree-sitter-cpp/blob/master/.github/workflows/fuzz.yml